pkg/api/v1/types.go

```diff
@@ -878,6 +879,8 @@ type PodSpec struct {
 	// If specified, these secrets will be passed to individual puller implementations for them to use.  For example,
 	// in the case of docker, only DockerConfig type secrets are honored.
 	ImagePullSecrets []LocalObjectReference `json:"imagePullSecrets,omitempty" description:"list of references to secrets in the same namespace available for pulling the container images"  patchStrategy:"merge" patchMergeKey:"name"`
+	// Optional: Conflicts are used to identify pods that cannot be scheduled together.
+	Conflicts [][]labels.Requirement `json:"conflicts,omitempty"`
```

Comment:
See also #341 and #7053. We need a LabelSelector type that has the right json, description, and patch tags.
This will also need description and patch tags.
Insert the field just below NodeSelector.
cc @sdminonne